src/Microsoft.ML.HalLearners/SymSgdClassificationTrainer.cs

```diff
@@ -155,25 +152,69 @@ public override TPredictor Train(TrainContext context)
 
         public override PredictionKind PredictionKind => PredictionKind.BinaryClassification;
 
-        public SymSgdClassificationTrainer(IHostEnvironment env, Arguments args)
-            : base(env, LoadNameValue)
+        /// <summary>
+        /// Initializes a new instance of <see cref="SymSgdClassificationTrainer"/>
+        /// </summary>
+        /// <param name="env">The private instance of <see cref="IHostEnvironment"/>.</param>
+        /// <param name="labelColumn">The name of the label column.</param>
+        /// <param name="featureColumn">The name of the feature column.</param>
+        /// <param name="weightColumn">The name for the column containing the initial weight.</param>
+        /// <param name="advancedSettings">A delegate to apply all the advanced arguments to the algorithm.</param>
+        public SymSgdClassificationTrainer(IHostEnvironment env, string featureColumn, string labelColumn,
+            string weightColumn = null, Action<Arguments> advancedSettings = null)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(LoadNameValue), TrainerUtils.MakeR4VecFeature(featureColumn),
+                  TrainerUtils.MakeR4ScalarLabel(labelColumn), TrainerUtils.MakeR4ScalarWeightColumn(weightColumn))
+        {
+            var args = new Arguments();
+
+            //apply the advanced args, if the user supplied any
+            args.Check(Host);
+            advancedSettings?.Invoke(args);
+            args.FeatureColumn = featureColumn;
+            args.LabelColumn = labelColumn;
+            // TODO:
+            //args.WeightColumn = weightColumn;
+
+            _args = args;
+            Info = new TrainerInfo();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="SymSgdClassificationTrainer"/>
+        /// </summary>
+        internal SymSgdClassificationTrainer(IHostEnvironment env, Arguments args)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(LoadNameValue), TrainerUtils.MakeR4VecFeature(args.FeatureColumn),
+                  TrainerUtils.MakeR4ScalarLabel(args.LabelColumn), TrainerUtils.MakeR4ScalarWeightColumn(null))
```

Comment:
>MakeR4ScalarLabel [](start = 31, length = 17)

bool

Reply:
You need a scalar bool label for binary classification. 

---
In reply to: [220057042](https://github.com/dotnet/machinelearning/pull/1012#discussion_r220057042) [](ancestors = 220057042)

Reply:
Thanks for the catch, I had only modified it above and forgot this instance. 

---
In reply to: [220299186](https://github.com/dotnet/machinelearning/pull/1012#discussion_r220299186) [](ancestors = 220299186,220057042)